reps/2023-01-09-aws-fleet-support.md

```diff
@@ -0,0 +1,138 @@
+
+### How will EC2 fleets clean up happen?
+
+There will be no change in how the instances are terminated today. However, terminating all the instances in a fleet does not necessarily delete the fleet. The fleet entity continues to be in active state. Hence, a `post_process` step will be introduced after each autoscaler update which cleans up any unused active fleets. We cannot describe fleets of `instant` type via EC2 APIs, hence it must be stored in either GCS kv store or any in-memory storage accessible to autoscaler.
+
+### How will caching stopped nodes work?
+
+When stopped nodes are cached, they are reused next time the cluster is setup. There is no change in this behavior w.r.t EC2 fleets. Any changes in instance states within a fleet does not affect fleet state based on the POC. Hence, they have to be explicitly deleted as part of `post_process`.
+
+### How will customer experience looks like?
+
+The autoscaler config or the customer experience with respect to provisioning workers using EC2 `create_instances` API will not change. An optional config key `node_launch_method` (note the naming convention) will be introduced within the `node_config` that determines whether to use `create_fleet` API or `create_instances` API while provisioning EC2 instances. However, default value for this config key will be `create_instances`. Any other arguments within the `node_config` will be passed to the appropriate API as is. However, the request type will be overridden to `instant` as Ray autoscaler cannot allow EC2 fleets to interfere with the scaling.
+
+```yaml
+....
+ray.worker.default:
+    max_workers: 100
+    node_config:
+      node_launch_method: create_fleet
+      OnDemandOptions:
+        AllocationStrategy: lowest-price
+      LaunchTemplateConfigs:
+        - LaunchTemplateSpecification:
+            LaunchTemplateName: RayClusterLaunchTemplate
+            Version: $Latest
+          Overrides:
+            - InstanceType: r3.8xlarge
+              ImageId: ami-04af5926cc5ad5248
+            - InstanceType: r4.8xlarge
+              ImageId: ami-04af5926cc5ad5248
+            - InstanceType: r5.8xlarge
+              ImageId: ami-04af5926cc5ad5248
+      TargetCapacitySpecification:
+        DefaultTargetCapacityType: 'on-demand'
+....
+```
+
+### What are the known limitations?
+
+-   The fleet request types `maintain` and `request` are not supported. Also, a hard limit on number of fleets of type `maintain` or `request` may block us from supporting that in future as well.
+-   EC2 fleet cannot span different subnets from same availability zone. This is more of a limitation of EC2 fleets which could impact availability.
```

Comment:
Just to clarify, we can still schedule across different subnets in _different_ AZs right? (that's also the standard ray setup)

Reply:
Yes, that's correct.